lib/buffer.js

```diff
@@ -1314,10 +1316,23 @@ function atob(input) {
   return Buffer.from(input, 'base64').toString('latin1');
 }
 
+function isUtf8(input) {
+  if (isTypedArray(input) || Buffer.isBuffer(input)) {
+    return bindingIsUtf8(input.buffer);
+  }
+
+  if (isAnyArrayBuffer(input)) {
+    return bindingIsUtf8(input);
+  }
+
+  return false;
```

Comment:
For consistency with other APIs shouldn't it throw an `ERR_INVALID_ARG_TYPE` error?

Reply:
Throwing errors come with a cost. My personal preference is towards returning `false`, but if this is the preferred way, I'll be happy to change it

Reply:
I think we should throw in this case. Otherwise, it will give a false impression the input provided has been validated and the result is: no utf-8, when in fact, the input is invalid.

Also, this branch will be rarely called.

Reply:
I agree. `Buffer.isUtf8('hello')` would return `false` which would be surprising.